qiskit/synthesis/unitary/qsd.py

```diff
@@ -106,6 +108,12 @@ def qs_decomposition(
            n-Qubit Gates Based on Block ZXZ-Decomposition*,
            `arXiv:2403.13692 <https://arxiv.org/abs/2403.13692>`_
     """
+    if (decomposer_1q is None or isinstance(decomposer_1q, OneQubitEulerDecomposer)) and (
+        decomposer_2q is None or isinstance(decomposer_2q, TwoQubitBasisDecomposer)
+    ):
+        array = np.asarray(mat, dtype=complex)
+        return QuantumCircuit._from_circuit_data(qsd.qs_decomposition(array, opt_a1, opt_a2))
```

Comment:
You are missing the `decomposer_1q` and `decomposer_2q` arguments on the `qsd.qs_decomposition` call here. We've validated they are none or the correct type here, so we need to make sure that we pass the user specified arg to the rust code. Right now this is ignoring the user specified decomposers.